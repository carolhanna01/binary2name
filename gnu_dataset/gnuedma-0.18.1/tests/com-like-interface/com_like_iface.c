/*
 * EDMA: Entorno de Desarrollo Modular y Abierto
 * Object Oriented and Componetware Framework
 * Copyright (C) 1998, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2010
 *    David Martínez Oliveira
 *
 * This file is part of EDMA.
 *
 * EDMA is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * EDMA is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with EDMA.  If not, see <http://www.gnu.org/licenses/>.
 */

/*
 * Simple example about how to map a single object in a way similar to
 * XX-COM systems.
 *
 * We use GNU/EDMA edma_new_obj as a generic object factory
 */
#include <edma.h>

typedef struct HELLO_WORLD_iface_t
{
  ESint32        (*born)(OBJID);
  ESint32        (*rip)(OBJID);
  EPVoid         (*say)(OBJID);
} IHELLO_WORLD;

int 
main ()
{
  OBJID        id;
  IHELLO_WORLD *iface;
  int          n;

  EDMAInit();

  id = edma_new_obj ("HELLO_WORLD");

  n = edma_get_all_met_func (id, (PPROC**)&iface);
  edma_printf ("Object %ld of class 'HELLO_WORLD' has %d methods",
	       id, n);

  /* For data member access use GNU/EDMA default accessors */
  edma_set_prop_sint32 (id, "num", 123);
  edma_set_prop_strz (id, "str", "Hello World!!!");
  iface->say(id);
  
  edma_free_obj (id);
  EDMAEnd();

}
