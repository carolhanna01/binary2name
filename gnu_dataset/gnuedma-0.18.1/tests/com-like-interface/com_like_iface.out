
GNU/EDMA 0.16.4 build [Jan 30 2010][Beta][0.16.4] starting up...
EDMA Startup process... OK
(HELLO_WORLD)  [Constructor]
Object 0 of class 'HELLO_WORLD' has 3 methods
(HELLO_WORLD)  [say] -------------------------------
(HELLO_WORLD)  [say] Hello world!!!!
(HELLO_WORLD)  [say] Property num : 123
(HELLO_WORLD)  [say] Property str : Hello World!!!
(HELLO_WORLD)  [say] -------------------------------
(HELLO_WORLD)  [Destructor]
---------------------[EDMAEnd]----------------------------------
No more process in GNU EDMA. Freeing All Resources...
Cleanup done!!
